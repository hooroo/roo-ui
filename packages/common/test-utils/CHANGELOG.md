# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.4.0"></a>
# [0.4.0](https://github.com/hooroo/roo-ui/compare/v0.3.3...v0.4.0) (2018-04-26)


### Features

* add theming ([bee896a](https://github.com/hooroo/roo-ui/commit/bee896a))
