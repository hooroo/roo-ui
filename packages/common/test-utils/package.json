{
  "name": "@roo-ui/test-utils",
  "version": "0.0.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
