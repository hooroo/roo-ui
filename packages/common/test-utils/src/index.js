export { default as shallowWithTheme } from './shallowWithTheme'; // eslint-disable-line import/prefer-default-export
