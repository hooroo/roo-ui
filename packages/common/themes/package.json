{
  "name": "@roo-ui/themes",
  "version": "0.4.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist --copy-files"
  }
}
