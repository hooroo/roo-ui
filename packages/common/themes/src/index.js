import qantas from './qantas';

export { qantas }; // eslint-disable-line import/prefer-default-export
