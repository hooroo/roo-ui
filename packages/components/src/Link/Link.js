import styled from 'styled-components';
import tag from 'clean-tag';
import { color, hover } from 'styled-system';

const Link = styled(tag.a)`
  cursor: pointer;
  text-decoration: none;

  ${color}
  ${hover}
`;

Link.propTypes = {
  ...color.propTypes,
  ...hover.propTypes,
};

Link.defaultProps = {
  color: 'ui.link',
  hover: {
    color: 'ui.linkHover',
  },
};

export default Link;
