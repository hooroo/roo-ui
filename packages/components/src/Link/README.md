# Link

<!-- STORY -->

## Installation

```shell
$ yarn add @roo-ui/components
```

## Example

```js
import { Link } from '@roo-ui/components';

export default (
  <Link>Hello world</Link>
);
```

### With [React Router](https://reacttraining.com/react-router/)

```js
import { Link as RouterLink } from 'react-router-dom';
import { Link as StyledLink } from '@roo-ui/components';

const Link = StyledLink.withComponent(RouterLink);

export default (
  <Link>Hello world</Link>
);
```

## Properties

| Name     | Description        | Type   | Default | Required? |
|:---------|:-------------------|:-------|:--------|:----------|
| `inline` | inherit font color | `bool` | false   | -         |

## Customization

This component can be customized with [styled-system](https://github.com/jxnblk/styled-system) by passing props for [color](https://github.com/jxnblk/styled-system#color-responsive), [space](https://github.com/jxnblk/styled-system, [hover](https://github.com/jxnblk/styled-system#hover), or [font weight](https://github.com/jxnblk/styled-system#typography).
