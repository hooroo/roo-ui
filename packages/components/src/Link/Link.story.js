import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';

import Link from '.';
import README from './README.md';

storiesOf('Components|Link', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Link href="https://www.qantas.com" target="_blank">Hello world</Link>
  ));
