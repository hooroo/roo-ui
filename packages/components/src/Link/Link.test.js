import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import Link from '.';

describe('<Link />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Link href="/">Hello world</Link>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
