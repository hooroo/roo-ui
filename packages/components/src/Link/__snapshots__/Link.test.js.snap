// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> renders correctly 1`] = `
.c0 {
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #e00e00;
}

.c0:hover {
  color: #870000;
}

<Clean.a
  className="c0"
  color="ui.link"
  hover={
    Object {
      "color": "#870000",
    }
  }
  href="/"
>
  Hello world
</Clean.a>
`;
